Bump the otel group with 4 updates and change builder functions

yeongjukang opened the pull request:
Hi, thank you for building this.
[This action](https://github.com/kube-rs/controller-rs/actions/runs/12474563589/job/34816752198) of #114 seems broken so I updated libs and changed builder functions following this OTEL's official guide below.
- https://github.com/open-telemetry/opentelemetry-rust/pull/2221

Additionally, *with_config* and *with_resource* functions for trace configuration will become private soon.
May I replace it with with_resource?

clux commented:
Hey, thanks a lot for trying this!

It's currently failing ci (ci tests both default features and `--all-features`), but happy to accept an upgrade that compiles and works. I'll try to push the approve and run  CI button as soon as possible.

> Additionally, with_config and with_resource functions for trace configuration will become private soon.
May I replace it with with_resource?

Go for it 🙏

yeongjukang commented:
@clux
Sorry I am beginner to Rust so didn't understand toolchains and rules properly.
I will try to make things polished like deprecated OTEL ports at justfile.
Thank you for the review and please give me some more few days!

yeongjukang commented:
@clux 
few changes added after first one
- changed deprecated port number from 55680 to 4317 for local test
- added docker-compose.yaml for local test and added run command at justfile
- with_resource applied

clux commented on justfile:
```diff
@@ -12,7 +12,8 @@ generate:
 # run with opentelemetry
 run-telemetry:
-  OPENTELEMETRY_ENDPOINT_URL=http://127.0.0.1:55680 RUST_LOG=info,kube=debug,controller=debug cargo run --features=telemetry
+  docker-compose up -d
```
nice. i might end up taking this out as i've killed off docker-compose deps elsewhere in favor of k3d installs, but i'll at least merge it in now for git history.

clux reviewed (approved):
Thanks a lot!

I see it builds and passes CI so that's proof enough for me.
